walletnode/services/artworkregister/task.go

```diff
@@ -530,9 +530,9 @@ func (task *Task) createArtTicket(_ context.Context) error {
 	// TODO: fill all 0 and "TBD" value with real values when other API ready
 	ticket := &pastel.ArtTicket{
 		Version:   1,
-		Author:    pastelID,
+		Author:    string(pastelID),
 		BlockNum:  task.artistBlockHeight,
-		BlockHash: task.artistBlockHash,
+		BlockHash: string(task.artistBlockHash),
```

Comment:
this code is not safe.
`task.artistBlockHash, err = hex.DecodeString(blockInfo.Hash)` 
based on this then task.artistBlockHash is bytes, not sure is representation of a string. Casting it to string are not safe.

Reply:
Thanks for raising. As I mentioned above, I had to change these data types to string because unmarshal was failing on []byte, I don't think changing it to string is unsafe, we can go back and forth from string to []byte safely. please share if you have a reference. Also, If you think there is a better way, please let me know, I'll be happy to make changes. 
I know there is one way of declaring our new data type and writing a custom unmarshal function for it which does hex decoding but that will introduce ripple effects and I believe string would work fine. 
please share your  solution proposal.

Reply:
sorry, I wrong used my personal account.
You can convert a bytes to string - only if ensure bytes are encode of a correct string
because if not - you will meet some problem later. I saw it many times.
for example - you can try with follow program - and see how it returns:
```
package main

import (
	"fmt"
)

func main() {
	   a := []byte{0xff}
    s := string(a)
    fmt.Println(s)
    for _, r := range s {
        fmt.Println(r)
    }
    rs := []rune(s)
    fmt.Println(rs)
}

```

Reply:
@pjenys0  I'm not disagreeing with you. What' I'm saying is; we can unmarshal it into string only that's already established. 
after that, we have many options,..either we can hex decode it (use ecnodeArtticket & decodeArtticket functions) or some other way. Please read this:

https://stackoverflow.com/questions/65872616/go-illegal-base64-data-at-input-byte-0-when-unmarshalling-a-string-json

so, I do not disagree, I'm saying that the fix would be in the scope of a separate PR which only addresses this issue by following the entire flow.. because of the importance of this, it should be a separate PR. 
what do you think?

Reply:
why need another PR?

Reply:
if so, why you don't resolve only decode issue in this PR only?

Reply:
decode issue is resolved that's why APIs are working fine. I've attached screenshots of testing and working PRs above in the other thread, Please take a look.

Reply:
mock-networks in current state is returning wrong hashes which fails hex decode. 
If I update mock-networks now then other changes will come in as well (what ayk mentioned today morning) and that will break the entire thing again. 
we agreed on slack that we'll work on that once we merge these PRs

Reply:
anw. I don't want to take time on this now.
So please resolve below command of @danileng20 . Then I agree to your solution now. But - then please create PR to fix it later.
In mean while - please help to review https://github.com/pastelnetwork/gonode/pull/147

Reply:
Sure, reviewing now